src/Web.DataAccess/Db/Commands/BaseDbCommand.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace Byndyusoft.Dotnet.Core.Samples.Web.DataAccess.Db.Commands
+{
+    using System;
+    using Infrastructure.Dapper;
+    using Infrastructure.Dapper.SessionsFactory;
+    using Microsoft.Extensions.Options;
+
+    public abstract class BaseDbCommand
+    {
+        protected readonly ISessionsFactory SessionsFactory;
+        protected readonly SqlExecutionOptions SqlExecutionOptions;
+
+        protected BaseDbCommand(ISessionsFactory sessionsFactory, IOptions<SqlExecutionOptions> sqlExecutionOptions)
+        {
+            if (sqlExecutionOptions == null)
+                throw new ArgumentNullException(nameof(sqlExecutionOptions));
+
+
+            SessionsFactory = sessionsFactory ?? throw new ArgumentNullException(nameof(sessionsFactory));
+            SqlExecutionOptions = sqlExecutionOptions.Value;
```

Comment:
Yep, IOptionsSnapshot can do this (https://andrewlock.net/reloading-strongly-typed-options-in-asp-net-core-1-1-0/), but I assume that it works only for Transient dependencies.